sessionctx/variable/session.go

```diff
@@ -548,6 +548,24 @@ func validateReadConsistencyLevel(val string) error {
 	}
 }
 
+// GetUserVar get the value and type of user variables.
+func (s *SessionVars) GetUserVar(name string) (types.Datum, bool, *types.FieldType, bool) {
+	dt, ok1 := s.userVars.vars[name]
+	ft, ok2 := s.userVars.types[name]
+	return dt, ok1, ft, ok2
+}
+
+// SetUserVar set the value and type of user variables.
+func (s *SessionVars) SetUserVar(name string, dt types.Datum, ft *types.FieldType) {
+	s.userVars.vars[name] = dt
+	s.userVars.types[name] = ft
+}
+
+// SetUserVarType is only used by rewriteVariable
+func (s *SessionVars) SetUserVarType(name string, ft *types.FieldType) {
```

Comment:
Better to remove this function since one purpose of this PR is to enforce the caller set `val` and `type` both for safety.